src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -37,8 +37,39 @@ public function process(ContainerBuilder $container)
         try {
             parent::process($container);
 
-            foreach ($this->unusedBindings as list($key, $serviceId)) {
-                $message = sprintf('Unused binding "%s" in service "%s".', $key, $serviceId);
+            foreach ($this->unusedBindings as list($key, $serviceId, $bindingType, $file)) {
+                $message = 'You have a "bind" configured for an argument ';
+
+                $argumentType = $argumentName = null;
+
+                if (false !== strpos($key, ' ')) {
+                    list($argumentType, $argumentName) = explode(' ', $key, 2);
+                } elseif ('$' === $key[0]) {
+                    $argumentName = $key;
+                } else {
+                    $argumentType = $key;
+                }
+
+                if ($argumentType) {
+                    $message .= sprintf('of type "%s" ', $argumentType);
+                }
+
+                if ($argumentName) {
+                    $message .= sprintf('named "%s" ', $argumentName);
+                }
+
+                if ($bindingType) {
+                    $message .= BoundArgument::$messages[$bindingType];
+                } else {
+                    $message .= sprintf('for service "%s"', $serviceId);
+                }
+
+                if ($file) {
+                    $message .= sprintf(' in file "%s"', $file);
+                }
+
+                $message .= '. But, this argument was not found in any of the services it was applied to. It may be unused and can be removed, or it may have a typo.';
```

Comment:
When $serviceId is used, this part could be improved as there is only one.
I'd also suggest to make the first part a single sentence (... but ...) e.g:
`A binding is configured for arguments [...] but no corresponding arguments have been found. It may be unused and should be removed, or it may have a typo.`

Reply:
The exception is thrown inside the foreach loop, so it actually contains information about a single binding. Should I change it to include all? What if there is a large amount of them?